forge-game/src/main/java/forge/game/staticability/StaticAbilityPanharmonicon.java

```diff
@@ -107,6 +107,19 @@ public static boolean applyPanharmoniconAbility(final StaticAbility stAb, final
             if (table.filterCards(origin == null ? null : ImmutableList.of(ZoneType.smartValueOf(origin)), ZoneType.smartValueOf(destination), stAb.getParam("ValidCause"), card, stAb).isEmpty()) {
                 return false;
             }
+        } else if (trigMode.equals(TriggerType.Attacks)) {
+            if (stAb.hasParam("ValidCause")) {
```

Comment:
Thanks!

`matchesValidParam` will already be true if the stAb doesn't have it, so you could avoid these hasParam checks